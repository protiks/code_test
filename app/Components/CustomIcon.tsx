import React from 'react';
import { IconButton } from '@mui/material';
import PictureAsPdfIcon from '@mui/icons-material/PictureAsPdf';
import InfoIcon from '@mui/icons-material/Info';
import OpenNewWindow from '../../utility/OpenNewWindow';

interface Props {
    url: string
    icon: string
}

const CustomIcon = ({ url, icon }: Props) => {
    const [hover, setHover] = React.useState(false)

    const handleColor = () => {
        return hover ? "info" : "action"
    }


    const renderSwitch = (param: string) => {
        switch (param) {
            case "pdf":
                return <PictureAsPdfIcon color={handleColor()} />
            case "info":
                return <InfoIcon color={handleColor()} />
            default:
                <div />
        }
    }

    return (
        <IconButton onClick={OpenNewWindow(url)} onMouseLeave={() => setHover(false)} onMouseEnter={() => setHover(true)}>
            {renderSwitch(icon)}
        </IconButton>
    )
}

export default CustomIcon;