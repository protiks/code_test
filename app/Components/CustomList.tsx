import React, { useState, useEffect } from "react";
import List from '@mui/material/List';
import { ListItemButton, ListItemText } from '@mui/material'
import ListItemIcon from '@mui/material/ListItemIcon';
import Divider from '@mui/material/Divider';
import ExpandLess from '@mui/icons-material/ExpandLess';
import ExpandMore from '@mui/icons-material/ExpandMore';
import Collapse from '@mui/material/Collapse';
import { Container } from '@mui/material';
import CustomIcon from "./CustomIcon";
import LimitStringLength from "../../utility/LimitStringLength";


interface Props {
    stuff: {
        liveUrl: string
        logoLightUrl: string
        description: string
        companyName: string
        companyId: number
        colorSettings: {
            brandColor: string
        }
        events: [
            {
                pdfUrl: string,
                reportUrl: string
            }
        ]

    }
}

const CustomList = ({ ...stuff }: Props) => {
    const [open, setOpen] = React.useState(false)
    const [r, setR] = React.useState('')
    const { logoLightUrl, companyName, liveUrl, description, events } = stuff.stuff

    useEffect(() => {
        if (events) {
            const [{ reportUrl }] = events
            setR(reportUrl)
        }
    }, events)

    const handleClick = () => {
        setOpen(!open);
    }

    return (
        <Container>
            <List >
                <ListItemButton onClick={handleClick}>
                    <ListItemIcon>
                        <img src={logoLightUrl} width={50} />
                    </ListItemIcon>
                    <ListItemText
                        primary={companyName}
                        secondary={LimitStringLength(description, !open)}
                    />
                    {open ? <ExpandLess /> : <ExpandMore />}
                </ListItemButton>
                <Collapse in={open} timeout="auto" unmountOnExit>
                    <List>
                        <ListItemIcon style={{ marginLeft: 50 }}>
                            <CustomIcon url={r} icon="pdf" />
                            <CustomIcon url={liveUrl} icon="info" />
                        </ListItemIcon>
                    </List>
                </Collapse>
                <Divider />
            </List>
        </Container>
    )
}


export default CustomList 