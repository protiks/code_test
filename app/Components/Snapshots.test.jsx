import renderer from 'react-test-renderer'
import CustomList from "./CustomList"
import Home from "../page"
import CustomIcon from './CustomIcon'

const data = [{
    stuff: {
        liveUrl: 'string',
        logoLightUrl: 'string',
        description: 'string',
        companyNawame: 'string',
        companyId: 1,
        colorSettings: {
            brandColor: 'string'
        },
        events: [
            {
                pdfUrl: 'string',
                reportUrl: 'string'
            },
            {
                pdfUrl: 'string',
                reportUrl: 'string'
            }
        ]

    }
}]

describe('something', () => {

    it('renders Custom List', () => {
        const tree = renderer.create(<CustomList stuff={data} />).toJSON()
        expect(tree).toMatchSnapshot()
    })

    it('renders Home Component', () => {
        const tree = renderer.create(<Home />).toJSON()
        expect(tree).toMatchSnapshot()
    })

    it('renders Custom Icon', () => {
        const tree = renderer.create(<CustomIcon />).toJSON()
        expect(tree).toMatchSnapshot()
    })

})