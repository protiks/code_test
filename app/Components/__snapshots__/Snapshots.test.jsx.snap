// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`something renders Custom Icon 1`] = `
<button
  className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
/>
`;

exports[`something renders Custom List 1`] = `
<div
  className="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
>
  <ul
    className="MuiList-root MuiList-padding css-h4y409-MuiList-root"
  >
    <div
      className="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-16ac5r2-MuiButtonBase-root-MuiListItemButton-root"
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
      >
        <img
          width={50}
        />
      </div>
      <div
        className="MuiListItemText-root css-tlelie-MuiListItemText-root"
      >
        <p
          className="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-83ijpv-MuiTypography-root"
        />
      </div>
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="ExpandMoreIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
        />
      </svg>
    </div>
    <hr
      className="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
    />
  </ul>
</div>
`;

exports[`something renders Home Component 1`] = `
[
  <div
    className="MuiBox-root css-i9gxme"
  >
    <header
      className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-1raotmp-MuiPaper-root-MuiAppBar-root"
    >
      <div
        className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-12tmiki-MuiToolbar-root"
      >
        <h6
          className="MuiTypography-root MuiTypography-h6 css-8dzdir-MuiTypography-root"
        >
          Quartr
        </h6>
        <button
          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1ei1svh-MuiButtonBase-root-MuiIconButton-root"
          disabled={false}
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="MenuIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
          <span
            className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </header>
  </div>,
  <div
    style={
      {
        "alignItems": "center",
        "display": "flex",
        "height": "50vh",
        "justifyContent": "center",
        "margin": "auto",
        "textAlign": "center",
        "width": "50vw",
      }
    }
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiLoadingButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiLoadingButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-6m3mwv-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
      disabled={false}
      id=":r0:"
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="fetch"
      tabIndex={0}
      type="button"
    >
      Fetch data
      <span
        className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>,
]
`;
